src/encoded/static/components/matrix_mouse_development.js

```diff
@@ -155,6 +155,43 @@ function sortMouseArray(a, b) {
     return aNumerical - bNumerical;
 }
 
+// Generate object listing all ages corresponding to a mouse stage based on rowCategoryData
+const getMouseAgeObject = (rowCategoryData) => {
+    const mouseAgeFullArray = [];
+    rowCategoryData.forEach((datum) => {
+        datum.life_stage_age.buckets.forEach((bucket) => {
+            mouseAgeFullArray.push(bucket.key.replace('embryonic', 'embryo'));
+        });
+    });
+    mouseAgeFullArray.sort(sortMouseArray);
+    const uniqueAgeArray = [...new Set(mouseAgeFullArray)];
+    const mouseAgeObject = {
+        embryo: uniqueAgeArray.filter(age => age.includes('embryo')).map(age => age.replace('embryo', '')),
+        postnatal: uniqueAgeArray.filter(age => age.includes('postnatal')).map(age => age.replace('postnatal', '')),
+        adult: uniqueAgeArray.filter(age => age.includes('adult')).map(age => age.replace('adult', '')),
+    };
+    return mouseAgeObject;
+};
+
+// Convert array of filters into search query
+// For life stage filters, we must append search terms for all corresponding ages
+const generateFilterQuery = (stageFilter, mouseAgeObject) => {
+    let selectedFilters = '';
+    if (stageFilter) {
+        stageFilter.forEach((f) => {
+            if (['adult', 'postnatal', 'embryo'].includes(f)) {
+                mouseAgeObject[f].forEach((mouseAge) => {
+                    selectedFilters += `life_stage_age=${f.replace('embryo', 'embryonic')}${mouseAge}&`;
```

Comment:
Like a previous review, string are immutable so its best to store values in an array and then do a join. However, its not a big deal for small loop-sizes